packages/sdk/src/impl/data/repositories/GenerateOffersRepositoryImpl.ts

```diff
@@ -45,12 +45,18 @@ export default class GenerateOffersRepositoryImpl
         return new VCLResult.Error(new VCLError("Offers did not returned."));
     }
 
-    parse(offersResponse: Response, token: VCLToken): VCLOffers {
+    parse(offersResponse: Response<JSONObject[] | { offers: JSONObject[], challenge?: string }>, token: VCLToken): VCLOffers {
+        const isLegacyOffersResponse = Array.isArray(offersResponse.payload);
+       
+        const offers = isLegacyOffersResponse ? offersResponse.payload : offersResponse.payload?.offers;
+        const challenge = isLegacyOffersResponse ? undefined : offersResponse.payload?.challenge;
+
         try {
             return new VCLOffers(
```

Comment:
Please see the `VCLOffers` data structure - https://github.com/velocitycareerlabs/WalletAndroid/blob/main/VCL/src/main/java/io/velocitycareerlabs/api/entities/VCLOffers.kt 
and
`GenerateOffersRepositoryImpl` implementation
https://github.com/velocitycareerlabs/WalletAndroid/blob/36dcc8df329c73fb02a84eacbc4bb34dc35d02da/VCL/src/main/java/io/velocitycareerlabs/impl/data/repositories/GenerateOffersRepositoryImpl.kt#L63